conda_build/utils.py

```diff
@@ -783,32 +780,13 @@ def _tar_xf_fallback(tarball, dir_path, mode="r:*"):
     from .os_utils.external import find_executable
 
     if tarball.lower().endswith(".tar.z"):
-        uncompress = find_executable("uncompress")
-        if not uncompress:
-            uncompress = find_executable("gunzip")
+        uncompress = find_executable("uncompress") or find_executable("gunzip")
         if not uncompress:
-            sys.exit(
-                """\
-uncompress (or gunzip) is required to unarchive .z source files.
-"""
-            )
+            sys.exit("uncompress/gunzip is required to unarchive .z source files.")
         check_call_env([uncompress, "-f", tarball])
         tarball = tarball[:-2]
 
-    t = tarfile.open(tarball, mode)
-    members = t.getmembers()
-    for i, member in enumerate(members, 0):
-        if os.path.isabs(member.name):
-            member.name = os.path.relpath(member.name, "/")
-        cwd = os.path.realpath(os.getcwd())
-        if not os.path.realpath(member.name).startswith(cwd):
-            member.name = member.name.replace("../", "")
-        if not os.path.realpath(member.name).startswith(cwd):
-            sys.exit("tarball contains unsafe path: " + member.name + " cwd is: " + cwd)
-        members[i] = member
-
-    t.extractall(path=dir_path)
-    t.close()
+    extract(tarball, dest_dir=dir_path)
```

Comment:
This section of code appears to prevent unpacking of unsafe paths. Do we have tests or are we sure that deferring to cph covers the potential security issues this code accounts for?

Reply:
CPH delegates to CPS which includes a check for unsafe paths:

https://github.com/conda/conda-package-streaming/blob/168beb4da1c971e8c99cee2c06c4862e267f8d81/conda_package_streaming/extract.py#L32-L51

```python
def is_within_dest_dir(name):
    abs_target = os.path.realpath(os.path.join(dest_dir, name))
    prefix = os.path.commonpath((dest_dir, abs_target))
    return prefix == dest_dir

for tar_file, _ in stream:
    # careful not to seek backwards
    def checked_members():
        # from conda_package_handling
        for member in tar_file:
            if not is_within_dest_dir(member.name):
                raise exceptions.SafetyError(f"contains unsafe path: {member.name}")
            yield member

    try:
        tar_file.extractall(path=dest_dir, members=checked_members())
    except OSError as e:
        if e.errno == ELOOP:
            raise exceptions.CaseInsensitiveFileSystemError() from e
        raise
```